packages/airnode-node/src/workers/local-gateways/validation.ts

```diff
@@ -102,6 +120,170 @@ export function verifyHttpSignedDataRequest(
   return { success: true, encodedParameters, endpointId: validEndpointId };
 }
 
+const beacon = z.object({
+  airnodeAddress: z.string(),
+  endpointId: z.string(),
+  encodedParameters: z.string(),
+  encodedValue: z.string().optional(),
+  timestamp: z.string().optional(),
+  signature: z.string().optional(),
+});
+
+export const signOevDataBodySchema = z.object({
+  chainId: z.number().positive(),
+  dapiServerAddress: z.string(),
+  oevProxyAddress: z.string(),
+  updateId: z.string(),
+  bidderAddress: z.string(),
+  bidAmount: z.string(),
+  signedData: z.array(beacon),
+});
+
+type Beacon = z.infer<typeof beacon>;
+interface BeaconDecoded extends Required<Beacon> {
+  decodedValue: ethers.BigNumber;
+}
+
+export type ProcessSignOevDataRequestBody = z.infer<typeof signOevDataBodySchema>;
+
+export function dropInvalidBeacons(beacons: Beacon[]) {
+  const currentTime = new Date();
+
+  return beacons.reduce((result, beacon) => {
+    const goProcessing = goSync(() => {
```

Comment:
nit:
```suggestion
    const goParseBeacon = goSync(() => {
```

I dislike the name, but my suggestion is not great either... I like that all the verification is encapsulated in this single goSync though.

Reply:
What about `goBeaconValidation`? That sounds better to me.